File: Musicefy.Core/Theme/ThemeMode.cs
namespace Musicefy.Core.Theme
{
    /// <summary>
    /// Brightness mode — entirely separate from the palette choice (AppTheme).
    /// Mirrors Aniyomi/Mihon's ThemeMode model.
    /// </summary>
    public enum ThemeMode
    {
        /// <summary>Follow the Windows light/dark registry key.</summary>
        System,
        /// <summary>Force light mode.</summary>
        Light,
        /// <summary>Force dark mode.</summary>
        Dark,
        /// <summary>Dark mode with pure-black (#000000) surfaces (AMOLED).</summary>
        Amoled,
    }
}

File: src/core/theme/ThemeMode.cpp
#include "AppTheme.h"
#include "ThemeMode.h"

namespace mf::core::theme {

int appThemeToInt(AppTheme t) {
    return static_cast<int>(t);
}
AppTheme appThemeFromInt(int v) {
    if (v < 0 || v > static_cast<int>(AppTheme::Dynamic)) {
        return AppTheme::Default;
    }
    return static_cast<AppTheme>(v);
}

int themeModeToInt(ThemeMode m) {
    return static_cast<int>(m);
}
ThemeMode themeModeFromInt(int v) {
    if (v < 0 || v > static_cast<int>(ThemeMode::Amoled)) {
        return ThemeMode::System;
    }
    return static_cast<ThemeMode>(v);
}

} // namespace mf::core::theme
